tests/testlib.nix

```diff
@@ -257,7 +257,7 @@ rec {
 
   fcVlanIfaces = mapAttrs' (vlan: vid: {
     name = "eth${vlan}";
-    value = { vlan = vid; assignIP = true; };
+    value = { vlan = vid; assignIP = false; };
```

Comment:
Is this intentional? This looks like it might break a lot of tests and Hydra seems to agree.

Reply:
It turns out that `assignIP = true` also configures the /etc/hosts file, which is used for name resolution in a few tests. I've added extra handling for this and the tests which had problems now seem to pass.